dev/testnet/upgrade.sh

```diff
@@ -0,0 +1,412 @@
+#!/usr/bin/env bash
+
+NUM_NODE=4
+
+set -e
+set -o errexit
+set -a
+set -m
+
+# set -x # activate debugging
+
+source oec.profile
+PRERUN=false
+
+function killbyname_gracefully() {
+  NAME=$1
+  ps -ef|grep "$NAME"|grep -v grep |awk '{print "kill  "$2", "$8}'
+  ps -ef|grep "$NAME"|grep -v grep |awk '{print "kill  "$2}' | sh
+  echo "All <$NAME> killed gracefully!"
+}
+
+function build_exchain() {
+  version=$1
+  (cd ../.. && git checkout dev && git pull && git checkout $version && make install)
+  echo "exchaind version ////"
+  exchaind version
+}
+
+function get_latest_height() {
+  node=$1
+  port_map='{"val0":26657,"val1":26757,"val2":26857,"val3":26957,"rpc4":27057,"rpc5":27157}'
```

Comment:
In this case, I think we should use global variable map type to carry the parameters, because of that in some case we have many case to bootstrap. In this approach we could lose some flexable action on modifying, also introduce a complexity in maintaining repo.

for example : 

```shell
# global variables definition 

rpc_port_map=()
rpc_port_map+=('{"name":26657,"name":27657}')
```

Futher more maybe we could introduce an approach to manage all parameters in global scope to maintain them clearly

Reply:
Tks,that is a good advice ,and I has optimized this part in the recent commit.